src/model.js

```diff
@@ -0,0 +1,242 @@
+      console.log("Search returned", features.length, "results");
+      callback(null, features);
+    },
+    function (err) {
+      // console.log("err", err);
+      callback(err);
+    }
+  );
+}
+
+function buildQueries(options) {
+  let queries;
+  if (options.geometry) {
+    queries = [buildQuery(options, options.geometry)];
+  } else {
+    queries = [buildQuery(options)];
+  }
+  return queries;
+}
+
+// Translate a request from the GeoServices API into something Yelp will understand
+function buildQuery(options, geometry) {
+  // we don't want to modify the passed in options object because Koop will use that later
+  // also for some reason the Yelp lib chokes when there is a passed in callback so omit it
+  const query = _.omit(_.cloneDeep(options), "callback");
+  if (geometry || options.geometry) {
+    const bbox = geometry || options.geometry;
+    [query.longitude, query.latitude] = getCenter(bbox);
+    query.radius = getRadius(bbox);
+  } else if (!options.location) {
+    query.location = "St. Louis, MO";
+  }
+  query.term = setTerm(options);
+  const sort = setSort(options);
+  if (sort) {
+    query.sort_by = sort;
+  }
+
+  query.limit = 50;
+  return query;
+}
+
+/**
+ * Given a bounding box (assuming web mercator right now), finds the center
+ * and returns [longitude, latitude]
+ *
+ * @param {object} bbox
+ */
+function getCenter(bbox) {
+  const x = (bbox.xmax + bbox.xmin) / 2;
+  const y = (bbox.ymax + bbox.ymin) / 2;
+
+  return proj.forward([x, y]);
+}
+
+/**
+ * Returns distance in meters from center to outside corner of bounding box.
+ *
+ * @param {object} bbox
+ */
+function getRadius(bbox) {
+  const halfX = Math.abs(bbox.xmax - bbox.xmin);
+  const halfY = Math.abs(bbox.ymax - bbox.ymin);
+
+  const radius = Math.sqrt(Math.pow(halfX, 2) + Math.pow(halfY, 2));
+  let roundedRadius = Math.round(radius);
+  if (roundedRadius > YELP_API_MAX_RADIUS) {
+    console.log(
+      "Warning: Radius was larger than the allowed maximum from Yelp API."
+    );
+    roundedRadius = YELP_API_MAX_RADIUS;
+  }
+  return roundedRadius;
+}
+
+// This function parses the search term from a geoservices where clause
+// 'where term = 'restaurants' => restaurants
+function setTerm(options) {
+  if (!options.where) return null;
+  const match = options.where.match(/.+\s*=\s*'(.+)'/);
+  return match ? match[1] : null;
+}
+
+// This function handles the sort part of yelps query. Depending on the type of search we are doing
+// we may want to sort by distance, rating or best match
+function setSort(options) {
+  if (options.orderByFields) {
+    // there may be multiple sorts - we only assume the first one since yelp can only do one at a time.
+    let orderByFields = options.orderByFields.split(",")[0];
+
+    if (orderByFields.indexOf(" ASC") > -1) {
+      // Yelp API does not allow sorting by ascending order ("ASC").
+      // If "ASC" order is passed, do nothing.
+      return false;
```

Comment:
If you are going to use `return` inside your `if` block (which is preferred), there is no reason to use `else`.